src/amberscript/parser.cc

```diff
@@ -1368,6 +1513,9 @@ Result Parser::ParseValues(const std::string& name,
 
       v.SetIntValue(token->AsUint64());
     }
+    ++seg_idx;
```

Comment:
This seems to allow partial initialization because I don't see a check that all the segments are intialized. Just want to confirm it is intentional. I haven't noticed tests for it.

Reply:
Added test. The buffer code will reject a mismatched number of items currently.